From 687129a26c04e116e2da1f8e7fec75e1026d8e6b Mon Sep 17 00:00:00 2001
From: Ahmad Samir <a.samirh78@gmail.com>
Date: Tue, 7 Oct 2025 22:08:57 +0300
Subject: [PATCH 352/553] tst_QMap: port away from QPair

Pick-to: 6.8 6.5
Task-number: QTBUG-115841
Change-Id: I2c62ef92c16b296e20907d03aa2b6a31f291cf63
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
(cherry picked from commit cc6f4cf6b44a961f310b7e2fef372f80921ff398)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 tests/auto/corelib/tools/qmap/tst_qmap.cpp | 30 ++++++++++++----------
 1 file changed, 16 insertions(+), 14 deletions(-)

diff --git a/tests/auto/corelib/tools/qmap/tst_qmap.cpp b/tests/auto/corelib/tools/qmap/tst_qmap.cpp
index afed1b53840..21ac2ccc14f 100644
--- a/tests/auto/corelib/tools/qmap/tst_qmap.cpp
+++ b/tests/auto/corelib/tools/qmap/tst_qmap.cpp
@@ -1010,9 +1010,9 @@ void tst_QMap::lowerUpperBound()
         QCOMPARE(emptyConstMap.upperBound(1), emptyConstMap.constEnd());
         QVERIFY(!emptyConstMap.isDetached());
 
-        const QMap<int, QString> constMap { qMakePair(1, "one"),
-                                            qMakePair(5, "five"),
-                                            qMakePair(10, "ten") };
+        const QMap<int, QString> constMap { {1, "one"},
+                                            {5, "five"},
+                                            {10, "ten"} };
 
         QCOMPARE(constMap.lowerBound(-1).key(), 1);
         QCOMPARE(constMap.lowerBound(1).key(), 1);
@@ -1057,9 +1057,9 @@ void tst_QMap::lowerUpperBound()
     QCOMPARE(emptyConstMap.upperBound(1), emptyConstMap.constEnd());
     QVERIFY(!emptyConstMap.isDetached());
 
-    const QMultiMap<int, QString> constMap { qMakePair(1, "one"),
-                                             qMakePair(5, "five"),
-                                             qMakePair(10, "ten") };
+    const QMultiMap<int, QString> constMap { {1, "one"},
+                                             {5, "five"},
+                                             {10, "ten"} };
 
     QCOMPARE(constMap.lowerBound(-1).key(), 1);
     QCOMPARE(constMap.lowerBound(1).key(), 1);
@@ -1804,11 +1804,11 @@ void tst_QMap::equal_range()
 {
     {
         const QMap<int, QString> constMap;
-        QCOMPARE(constMap.equal_range(1), qMakePair(constMap.constEnd(), constMap.constEnd()));
+        QCOMPARE(constMap.equal_range(1), std::pair(constMap.constEnd(), constMap.constEnd()));
         QVERIFY(!constMap.isDetached());
 
         QMap<int, QString> map;
-        QCOMPARE(map.equal_range(1), qMakePair(map.end(), map.end()));
+        QCOMPARE(map.equal_range(1), std::pair(map.end(), map.end()));
 
         map.insert(1, "value1");
         map.insert(5, "value5");
@@ -1819,28 +1819,30 @@ void tst_QMap::equal_range()
         QCOMPARE(pair.second.value(), "value5");
         auto b = map.find(1);
         auto e = map.find(5);
-        QCOMPARE(pair, qMakePair(b, e));
+        QCOMPARE(pair, std::pair(b, e));
 
         pair = map.equal_range(3);
         QCOMPARE(pair.first.value(), "value5");
         QCOMPARE(pair.second.value(), "value5");
-        QCOMPARE(pair, qMakePair(e, e));
+        QCOMPARE(pair, std::pair(e, e));
 
-        QCOMPARE(map.equal_range(10), qMakePair(map.end(), map.end()));
+        QCOMPARE(map.equal_range(10), std::pair(map.end(), map.end()));
     }
 
     const QMultiMap<int, QString> constMap;
-    QCOMPARE(constMap.equal_range(1), qMakePair(constMap.constEnd(), constMap.constEnd()));
+    QCOMPARE(constMap.equal_range(1), std::pair(constMap.constEnd(), constMap.constEnd()));
     QVERIFY(!constMap.isDetached());
 
     QMultiMap<int, QString> map;
     const QMultiMap<int, QString> &cmap = map;
 
-    QPair<QMultiMap<int, QString>::iterator, QMultiMap<int, QString>::iterator> result = map.equal_range(0);
+    using It = QMultiMap<int, QString>::iterator;
+    std::pair<It, It> result = map.equal_range(0);
     QCOMPARE(result.first, map.end());
     QCOMPARE(result.second, map.end());
 
-    QPair<QMultiMap<int, QString>::const_iterator, QMultiMap<int, QString>::const_iterator> cresult = cmap.equal_range(0);
+    using CIt = QMultiMap<int, QString>::const_iterator;
+    std::pair<CIt, CIt> cresult = cmap.equal_range(0);
     QCOMPARE(cresult.first, cmap.cend());
     QCOMPARE(cresult.second, cmap.cend());
 
-- 
2.51.2

